samplesheets/tests/test_views_api_taskflow.py

```diff
@@ -250,6 +348,537 @@ def test_post_created(self):
+        }
+
+    def test_create(self):
+        """Test post() in IrodsAccessTicketCreateAPIView as admin"""
+        self.assertEqual(IrodsAccessTicket.objects.count(), 0)
+        self.assert_alert_count(CREATE_ALERT, self.user, 0)
+        self.assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self.assertEqual(self.get_tl_event_count('create'), 0)
+        self.assertEqual(self.get_app_alert_count('create'), 0)
+
+        with self.login(self.user):
+            response = self.client.post(self.url, self.post_data)
+
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(IrodsAccessTicket.objects.count(), 1)
+        ticket = IrodsAccessTicket.objects.first()
+        self.assertEqual(ticket.path, self.path)
+        self.assertEqual(ticket.label, self.label)
+        self.assertEqual(
+            ticket.date_expires,
+            datetime.strptime(self.date_expires, '%Y-%m-%dT%H:%M:%S.%f%z'),
+        )
+        self.assertEqual(ticket.user, self.user)
+        self.assertEqual(ticket.study, self.study)
+        self.assertEqual(ticket.assay, self.assay)
+        self.assert_alert_count(CREATE_ALERT, self.user, 0)
+        self.assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self.assertEqual(self.get_tl_event_count('create'), 1)
+        self.assertEqual(self.get_app_alert_count('create'), 2)
+        self.assertEqual(
+            self.app_alert_model.objects.filter(
+                alert_name='irods_ticket_create'
+            )
+            .first()
+            .user,
+            self.user_delegate,
+        )
+
+        # Assert ticket state in iRODS
+        irods_ticket = self.get_irods_ticket(ticket)
+        self.assertEqual(irods_ticket[TicketQuery.Ticket.type], 'read')
+        self.assertIsNotNone(irods_ticket[TicketQuery.Ticket.expiry_ts])
+        self.assertEqual(
+            irods_ticket[TicketQuery.Collection.name], self.coll.path
+        )
+
+    def test_create_contributor(self):
+        """Test post() in IrodsAccessTicketCreateAPIView as contributor"""
+        self.assertEqual(IrodsAccessTicket.objects.count(), 0)
+        self.assert_alert_count(CREATE_ALERT, self.user, 0)
+        self.assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self.assertEqual(self.get_tl_event_count('create'), 0)
+        self.assertEqual(self.get_app_alert_count('create'), 0)
+
+        with self.login(self.user_contrib):
+            response = self.client.post(self.url, self.post_data)
+
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(IrodsAccessTicket.objects.count(), 1)
+        ticket = IrodsAccessTicket.objects.first()
+        self.assertEqual(ticket.path, self.path)
+        self.assertEqual(ticket.label, self.label)
+        self.assertEqual(
+            ticket.date_expires,
+            datetime.strptime(self.date_expires, '%Y-%m-%dT%H:%M:%S.%f%z'),
+        )
+        self.assertEqual(ticket.user, self.user_contrib)
```

Comment:
The important thing to test here is the user and of course the created alerts and events. I think we can omit the assertions for other model properties as we already check them in `test_create()`.